modules/tinymce/src/core/main/ts/api/EditorOptions.ts

```diff
@@ -0,0 +1,277 @@
+  /**
+   * Register a new option that can be used within TinyMCE.
+   *
+   * @method register
+   * @param {String} name Name of an option.
+   * @param {OptionSpec} spec A option spec containing how to valid the option and other optional metadata.
+   */
+  register: {
+    <K extends BuiltInOptionType>(name: string, spec: BuiltInOptionSpec<K>): void;
+    <K extends keyof EditorSettings>(name: K, spec: OptionSpec<EditorSettings[K], EditorOptions[K]> | SimpleOptionSpec<EditorSettings[K]>): void;
+    <T, U>(name: string, spec: OptionSpec<T, U>): void;
+    <T>(name: string, spec: SimpleOptionSpec<T>): void;
+  };
+
+  /**
+   * Check if an option has been registered.
+   *
+   * @method isRegistered
+   * @param {String} name Name of an option.
+   * @return {Boolean} True if the option has already been registered, otherwise false.
+   */
+  isRegistered: (name: string) => boolean;
+
+  /**
+   * Get the value of a registered option.
+   *
+   * @method get
+   * @param {String} name Name of a registered option.
+   * @return {Object} An option value or the registered default if not set.
+   */
+  get: {
+    <K extends keyof EditorOptions>(name: K): EditorOptions[K] | undefined;
+    <T>(name: string): T | undefined;
+  };
+
+  /**
+   * Set the value for a registered option.
+   *
+   * @method set
+   * @param {String} name Name of a registered option.
+   * @return {Boolean} True if the option value was successfully set, otherwise false.
+   */
+  set: {
+    <K extends keyof EditorOptions>(name: K, value: EditorOptions[K]): boolean;
+    <T>(name: string, value: T): boolean;
+  };
+
+  /**
+   * Clear any set value for the specified option so that it'll revert back to using the default value.
+   *
+   * @method unset
+   * @param {String} name Name of a registered option.
+   * @return {Boolean} True if the option value was successfully reset, otherwise false.
+   */
+  unset: (name: string) => boolean;
+}
+
+const getBuiltInProcessor = <K extends BuiltInOptionType>(type: K): Processor<BuiltInOptionTypeMap[K], BuiltInOptionTypeMap[K]> => {
+  const validator = (() => {
+    switch (type) {
+      case 'array':
+        return Type.isArray;
+      case 'boolean':
+        return Type.isBoolean;
+      case 'function':
+        return Type.isFunction;
+      case 'number':
+        return Type.isNumber;
+      case 'object':
+        return Type.isObject;
+      case 'string':
+        return Type.isString;
+      case 'string[]':
+        return (val) => Type.isArrayOf(val, Type.isString);
+    }
+  })();
+
+  return (value) => ({ value, valid: validator(value), message: `The value must be a ${type}.` });
+};
+
+const isBuiltInSpec = <K extends BuiltInOptionType>(spec: unknown): spec is BuiltInOptionSpec<K> =>
+  Type.isString((spec as BuiltInOptionSpec<K>).processor);
+
+const getErrorMessage = (message: string, result: ProcessorError): string => {
+  const additionalText = !Strings.isEmpty(result.message) ? '. ' + result.message : '';
```

Comment:
```suggestion
  const additionalText = Strings.isEmpty(result.message) ? '' : '. ' + result.message;
```

Reply:
I'm not a fan of this as it makes it harder to read.

Reply:
I always find negated ternaries rough, and also like when the `?` and `:` are closer together

what about

```suggestion
  const additionalText = Strings.isEmpty(result.message) ? '' : `. ${result.message}`;
```

Reply:
That I can live with :+1: